ProcessLib/Process.h

```diff
@@ -100,6 +100,14 @@ class Process
         int const /*process_id*/)
     {
     }
+
+    virtual void extrapolateIntPtProcessSolutions(
+        const double /*t*/,
+        std::vector<GlobalVector*>& /*process_solutions*/,
+        std::vector<GlobalVector*> const /*transport_solutions*/)
```

Comment:
Not clear from the arguments' names who is who. Together with the function name it is not clear what is being extrapolated and what the result will be. Also same applies here, that in the general process interface there should be no names specific to some process.
A suggestion, though you can come up with a better names:
`extrapolateIntegrationPointValuesToNodes(t, integration_point_values_vectors, node_values_vectors)`

Also place the output variables at the end.

Second argument should be const&.

Reply:
Fixed.